### PR TITLE
feat: add documentation link in help messages

### PR DESCRIPTION
**Pull request by mr-cal**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox`?

-----

Adds documentation links in general help messages and command help messages.

### CODE REVIEW

File: tests/integration/test_application.py [hunk 1]
```diff
@@ -81,6 +81,7 @@ def app(create_app):
 
 For more information about a command, run 'testcraft help <command>'.
 For a summary of all commands, run 'testcraft help --all'.
+For more information about testcraft, check out: www.craft-app.com/docs/3.14159
```
Comment (lengau) [inline tests/integration/test_application.py:84]: Probably best not to point this to another site of unknown provenance. I've replaced with [RFC 2606](https://datatracker.ietf.org/doc/html/rfc2606) domains.
```suggestion
For more information about testcraft, check out: testcraft.example/docs/3.14159
```

File: tests/integration/test_application.py [hunk 2]
```diff
@@ -248,6 +249,10 @@ def test_get_command_help(monkeypatch, emitter, capsys, app, cmd, help_param):
     stdout, stderr = capsys.readouterr()
 
     assert f"testcraft {cmd} [options]" in stderr
+    assert stderr.endswith(
+        "For more information, check out: "
+        f"www.craft-app.com/docs/3.14159/reference/commands/{cmd}\n\n"
```
Comment (lengau) [inline tests/integration/test_application.py:254]:
```suggestion
        f"testcraft.example/docs/3.14159/reference/commands/{cmd}\n\n"
```

File: tests/unit/test_application.py [hunk 3]
```diff
@@ -2223,3 +2223,37 @@ def test_app_config_in_help(
...
+)
+@pytest.mark.usefixtures("emitter")
+def test_doc_url_in_general_help(help_args, monkeypatch, capsys, app):
+    """General help messages contain a link to the documentation."""
+    monkeypatch.setattr(sys, "argv", ["testcraft", *help_args])
+
+    with pytest.raises(SystemExit):
+        app.run()
+
+    expected = "For more information about testcraft, check out: www.craft-app.com/docs/3.14159\n\n"
```
Comment (lengau) [inline tests/unit/test_application.py:2243]:
```suggestion
    expected = "For more information about testcraft, check out: testcraft.example/docs/3.14159\n\n"
```

File: tests/unit/test_application.py [hunk 4]
```diff
@@ -2223,3 +2223,37 @@ def test_app_config_in_help(
...
+@pytest.mark.usefixtures("emitter")
+def test_doc_url_in_command_help(monkeypatch, capsys, app):
+    """Command help messages contain a link to the command's doc page."""
+    app.add_command_group("Test", [AppConfigCommand])
+    monkeypatch.setattr(sys, "argv", ["testcraft", "app-config", "-h"])
+
+    with pytest.raises(SystemExit):
+        app.run()
+
+    expected = "For more information, check out: www.craft-app.com/docs/3.14159/reference/commands/app-config\n\n"
```
Comment (lengau) [inline tests/unit/test_application.py:2257]:
```suggestion
    expected = "For more information, check out: testcraft.example/docs/3.14159/reference/commands/app-config\n\n"
```

**Review by lengau — APPROVED** [review]
LGTM, though we should probably use an RFC 2606 domain rather than craft-app.com (which is owned by someone other than Canonical).

**Comment by lengau** [comment]
Also I know the linter breakage is unrelated, but since an autoformat would fix it I think it's ok to include here :-)

**Comment by mr-cal** [comment]
> Also I know the linter breakage is unrelated, but since an autoformat would fix it I think it's ok to include here :-)

Interesting bug: https://github.com/astral-sh/ruff/issues/14228

**Comment by dariuszd21** [comment]
> > Also I know the linter breakage is unrelated, but since an autoformat would fix it I think it's ok to include here :-)
> 
> Interesting bug: [astral-sh/ruff#14228](https://github.com/astral-sh/ruff/issues/14228)

I used `-` instead of `#` as I found that it was somewhere in the codebase already

**Pull request merged**